src/Tasks/GetCompatiblePlatform.cs

```diff
@@ -60,60 +60,11 @@ public override bool Execute()
 
                 string projectReferencePlatformMetadata = AssignedProjectsWithPlatform[i].GetMetadata("Platforms");
```

Comment:
These metadata calls can be called from `PlatformNegotiation.GetPlatformString` instead of here. `PlatformNegotiation.GetPlatformString` is essentially the implementation of this task now, so you can remove all the metadata calls and just pass `PlatformNegotiation.GetPlatformString` the `AssignedProjectsWithPlatform` and extract metadata there.

Reply:
So I did some digging and debugging. there is no place in the graph build where both the project and the project it is referencing are both evaluated. so we need to evaluate the referenced project before we do the negotiation. this is similar to how "new TaskItem(AnnotatedProjects[i])" is run in getCompatablePlatform.cs(58).